content/techniques/caching.md

```diff
@@ -161,6 +161,42 @@ import { AppController } from './app.controller';
 export class ApplicationModule {}
 ```
 
+#### Redis cache `set` and `get` example
+
+First install the type dependency:
+
+```bash
+$ npm install --save-dev @types/cache-manager
+```
+
+```typescript
+// AppService.ts
+import { Injectable, Inject, CACHE_MANAGER } from '@nestjs/common';
+import { Cache } from 'cache-manager';
+
+@Injectable()
+export class AppService {
+  constructor(@Inject(CACHE_MANAGER) private cacheManager: Cache) {
+  }
+
+  async getHello(): Promise<string> {
+    // callback way
+    this.cacheManager.set('foo', 'bar', {ttl: 100000}, (err) => {
+      this.cacheManager.get('foo', (errGet, result) => {
+        console.log(result); // logs "bar"
+      });
+    });
+
+    // async/await way
+    await this.cacheManager.set('foo2', 'bar2', {ttl: 100000});
```

Comment:
It might be a good practice to wrap this in a try/catch block.

Reply:
Agree. Just for docs it is kept like this and trying to keep minimal. I checked other part of docs, not at all place `await`  statement are in try-catch blocks. E.g. https://docs.nestjs.com/pipes#class-validator

Reply:
Alright, makes sense.